include/swift/AST/DiagnosticsSema.def

```diff
@@ -2328,7 +2328,14 @@ WARNING(optional_check_promotion,none,
 WARNING(optional_pattern_match_promotion,none,
         "pattern match introduces an implicit promotion from %0 to %1",
         (Type, Type))
-
+WARNING(optional_to_any_coercion,none,
+        "expression implicitly coerced from %0 to Any", (Type))
+NOTE(default_optional_to_any,none,
+     "provide a default value to avoid the warning", ())
+NOTE(force_optional_to_any,none,
+     "force-unwrap the value to avoid the warning", ())
+NOTE(silence_optional_to_any,none,
+     "explicitly cast to Any with 'as Any' to silence the warning", ())
```

Comment:
Nitpick: Clang typically says *"this* warning" rather than *"the* warning". I don't think Swift has enough of these to count as having a convention.